src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,241 @@
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't create car " + car, throwables);
+        }
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT model, manufacturer_id FROM cars WHERE id = ? "
+                + "AND is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCar(resultSet);
+                car.setId(id);
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't get car from dataBase by id - "
+                    + id, throwables);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversFromCurrentCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT * FROM cars WHERE is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                Long carsId = resultSet.getObject("id", Long.class);
+                Car car = parseCar(resultSet);
+                car.setId(carsId);
+                car.setDrivers(getDriversFromCurrentCar(carsId));
+                cars.add(car);
+            }
+            return cars;
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't get all cars", throwables);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?,"
+                   + "manufacturer_id = ? WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                  PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't update current car - " + car,
+                    throwables);
+        }
+        if (car != null) {
```

Comment:
we don't need this check, if car is null you will get NPE earlier on car.getModel()

Reply:
Removed